### ISSUE TITLE
Support render hooks for specific templates only

### ISSUE THREAD

**Issue by Brixy** [gohugoio/hugo#7349]
Renderhooks are a great addition, and I think it does make sense to use image and link render hooks globally.

Yet, it may make sense to use heading render hooks occasionally—e.g. for blogs articles or other sections only.

would it be possible and make sense to enable a folder hierarchy like this:

```
layouts/blog/_markup/render-heading.html
```

Thank you!

**Comment by bep** [comment]
I agree, and I had it orignally implemented to follow mostly the same lookup rules as, say, `single.html`, but I restricted it the last moment after a comment from @regisphilibert. But there obviously is a need to have different set of hooks for blog vs docs etc.

**Comment by regisphilibert** [comment]
🤔  don't recall which comment... In any case I fully support having render hooks applied to certain types of content, and using the directory structure is a UX every Hugo user is familiar with.
